zerver/lib/message_cache.py

```diff
@@ -246,6 +246,10 @@ def finalize_payload(
         else:
             obj["content_type"] = "text/x-markdown"
 
+        for item in obj.get("edit_history", []):
+            if "prev_rendered_content_version" in item:
+                del item["prev_rendered_content_version"]
```

Comment:
Why here rather than `build_message_dict`?

Reply:
All such reformatting to turn the object into an API-format JSONable dictionary is done in **finalize_payload** .

Like:
```py
        if apply_markdown:
            obj["content_type"] = "text/html"
            obj["content"] = obj["rendered_content"]
        else:
            obj["content_type"] = "text/x-markdown"

        del obj["sender_realm_id"]
        del obj["sender_avatar_source"]
            
```
etc... 

Also, **finalize_payload** is directly used in `process_message_event` & `GenericOutgoingWebhookService.make_request` instead of `build_message_dict`

Reply:
OK. I guess the main trade-off is caching. But I guess since we're going to add content-type manipulation, and that'll need to be here, probably doing all the edit_history fiddling in one place is good.